cf_xarray/accessor.py

```diff
@@ -481,23 +482,80 @@ def _guess_bounds_dim(da, dim=None, out_dim="bounds"):
         elif len(dim) == 2:
             daX = _guess_bounds_dim(da, dim[0]).rename(bounds="Xbnds")
             daXY = _guess_bounds_dim(daX, dim[1]).rename(bounds="Ybnds")
-            return xr.concat(
-                [
-                    daXY.isel(Xbnds=0, Ybnds=0),
-                    daXY.isel(Xbnds=0, Ybnds=1),
-                    daXY.isel(Xbnds=1, Ybnds=1),
-                    daXY.isel(Xbnds=1, Ybnds=0),
-                ],
-                out_dim,
-            ).transpose(..., "bounds")
+            # At this point, we might have different corners for adjacent cells, we average them together to have a nice grid
```

Comment:
lets stick  in its own helper function (`_guess_bounds_2d`)

Reply:
Done! I also sent the functions to `helpers.py`.